ocean_lib/ocean/ocean_assets.py

```diff
@@ -443,3 +445,52 @@ def download_asset(
             index=index,
             userdata=userdata,
         )
+
+    @enforce_types
+    def pay_for_service(self, asset: Asset, service: Service, wallet: Wallet):
+        dt = ERC20Token(self._web3, service.data_token)
+        balance = dt.balanceOf(wallet.address)
+
+        if balance < to_wei(1):
+            raise InsufficientBalance(
+                f"Your token balance {pretty_ether_and_wei(balance, dt.symbol())} is not sufficient "
+                f"to execute the requested service. This service "
+                f"requires {pretty_ether_and_wei(1, dt.symbol())}."
+            )
+
+        consumable_result = service.is_consumable(
+            asset, {"type": "address", "value": wallet.address}
+        )
+        if consumable_result != ConsumableCodes.OK:
+            raise AssetNotConsumable(consumable_result)
+
+        data_provider = DataServiceProvider
+        initialize_response = data_provider.initialize(
+            did=asset.did,
+            service_id=service.id,
+            consumer_address=wallet.address,
+            service_endpoint=data_provider.build_initialize_endpoint(
+                self._config.provider_url
+            )[1],
+        )
+
+        tx_id = dt.start_order(
+            consumer=wallet.address,
+            service_id=int(service.id),
```

Comment:
I need to grab David's helper, this is not correct. We should also search for wherever we are doing this :)